### PR TITLE
[net] Don't return an optional from TransportDeserializer::GetMessage()

### PR DESCRIPTION
**Pull request by jnewbery**
Also, access mapRecvBytesPerMsgCmd with `at()` not `find()`. This
throws an error if COMMAND_OTHER doesn't exist, which should never
happen. `find()` instead just accessed the last element, which could make
debugging more difficult.

Resolves review comments from PR19107:

- https://github.com/bitcoin/bitcoin/pull/19107#discussion_r478718436
- https://github.com/bitcoin/bitcoin/pull/19107#discussion_r478714497

### CODE REVIEW

**Comment by jnewbery** [comment]
This is #20364, rebased on master.

Requesting rereview from @ryanofsky @MarcoFalke and @ajtowns, who reviewed the original PR.

**Comment by practicalswift** [comment]
Concept ACK

File: src/test/fuzz/p2p_transport_serialization.cpp [hunk 1]
```diff
@@ -68,18 +68,16 @@ FUZZ_TARGET_INIT(p2p_transport_serialization, initialize_p2p_transport_serializa
...
-            uint32_t out_err_raw_size{0};
-            std::optional<CNetMessage> result{deserializer.GetMessage(m_time, out_err_raw_size)};
-            if (result) {
-                assert(result->m_command.size() <= CMessageHeader::COMMAND_SIZE);
-                assert(result->m_raw_message_size <= mutable_msg_bytes.size());
-                assert(result->m_raw_message_size == CMessageHeader::HEADER_SIZE + result->m_message_size);
-                assert(result->m_time == m_time);
+            bool reject_message{false};
+            CNetMessage msg = deserializer.GetMessage(m_time, reject_message);
+            assert(msg.m_command.size() <= CMessageHeader::COMMAND_SIZE);
```
Comment (theStack) [inline src/test/fuzz/p2p_transport_serialization.cpp:73]: Just to get sure, is it intended that the block following the `GetMessage` is now not conditional any more? (i.e. within a `if (!reject_message) { ... }`)

**Review by theStack — COMMENTED** [review]
Concept ACK

File: src/test/fuzz/p2p_transport_serialization.cpp (hunk 1 above)
Comment (jnewbery) [reply src/test/fuzz/p2p_transport_serialization.cpp:73]: Yes, it's essentially reverting the change here: https://github.com/bitcoin/bitcoin/commit/890b1d7c2b8312d41d048d2db124586c5dbc8a49#diff-7f9b91d8cb2eaad7b4c0e1a9a4650d9a089c0908faa0502703f3b13a920558afR37, when the return type of `GetMessage()` was changed to an optional.

**Review by theStack — APPROVED** [review]
Code-review ACK f3e451bebfe2e2d8de901d8ac29c064a51d3b746

Also rebased on master + verified that the build is clean and all tests pass.

**Review by ryanofsky — APPROVED** [review]
Code review ACK f3e451bebfe2e2d8de901d8ac29c064a51d3b746. Changes since last review in https://github.com/bitcoin/bitcoin/pull/20364#pullrequestreview-534369904 were simplifying by dropping the third commit, rebasing, and cleaning up some style & comments in the first commit.

The cleanups here are all small and straightforward, but I think the best one is getting rid of the unchecked `map.find` iterator dereference, and getting rid of the redundant & sometimes initialized `out_err_raw_size` output parameter.

**Pull request merged**